### PR TITLE
fix(local-cli): Updated for VS2017

### PR DESCRIPTION
**Pull request by sayar**
Added support for VS2017 without requiring any modifications for VS2015.

### CODE REVIEW

File: local-cli/runWindows/utils/build.js [hunk 1]
```diff
@@ -22,8 +22,14 @@ function buildSolution(slnFile, buildType, buildArch, verbose) {
 }
 
 function restoreNuGetPackages(options, slnFile, verbose) {
+  let nugetPath = options.nugetPath || path.join(options.root, 'node_modules/react-native-windows/local-cli/runWindows/.nuget/nuget.exe');
+  const msBuildTools = MSBuildTools.findAvailableVersion();
+  if(msBuildTools.version === '15.0' && !options.nugetPath) {
+    console.log(chalk.green('Using NuGet version >v.4'));
+    nugetPath = path.join(options.root, 'node_modules/react-native-windows/local-cli/runWindows/.nuget/nuget.v4.1.0.exe');
```
Comment (rozele) [inline local-cli/runWindows/utils/build.js]: >v4.*

File: local-cli/runWindows/utils/msbuildtools.js [hunk 2]
```diff
@@ -57,19 +64,38 @@ class MSBuildTools {
...
       }
-      console.log(chalk.green(`Found MSBuild v${version} at ${toolsPath}`));
-      return new MSBuildTools(version, toolsPath);
+      toolsPath = toolsPathOutput;
     }
   } catch (e) {
+    toolsPath = null;
+  }
+  // Try to get the MSBuild Path by checking file system (for VS2017)
+  if (!toolsPath) {
```
Comment (rozele) [inline local-cli/runWindows/utils/msbuildtools.js]: Is there documentation on this approach?

File: local-cli/runWpf/utils/build.js [hunk 3]
```diff
@@ -22,12 +22,17 @@ function buildSolution(slnFile, buildType, buildArch, verbose) {
 }
 
 function restoreNuGetPackages(options, slnFile) {
+  let nugetPath = options.nugetPath || path.join(options.root, 'node_modules/react-native-windows/local-cli/runWpf/.nuget/nuget.exe');
+  const msBuildTools = MSBuildTools.findAvailableVersion();
+  if(msBuildTools.version === '15.0' && !options.nugetPath) {
+    console.log(chalk.green('Using NuGet version >v.4'));
```
Comment (rozele) [inline local-cli/runWpf/utils/build.js]: >>v.4 [](start = 49, length = 4)

>v4.*

File: local-cli/runWindows/utils/msbuildtools.js (hunk 2 above)
Comment (sayar) [inline local-cli/runWindows/utils/msbuildtools.js]: VS2017 removed registry variables so you can't locate VS the same way anymore. See: https://docs.microsoft.com/en-us/visualstudio/msbuild/what-s-new-in-msbuild-15-0

Also, I found this today: https://github.com/Microsoft/vswhere so I am going to switch to using that exe once I can package it up as a separate npm module that we can depend on.

**Comment by sayar** [comment]
@rozele I switched to using vswhere instead.

**Comment by sayar** [comment]
Also fixes #985 and does everything that #1069 does without the download nuget script and refactoring everything.

**Comment by AlanFoster** [comment]
❔ Does react-native-windows support 2017 now? If so, should the readme be updated now too? 🤔 

> System Requirements
> ...
>   - Download Visual Studio 2015 Community
>
> ...
>
>Note: Development on React Native Windows itself currently requires Visual Studio 2015. It is not supported with Visual Studio 2017 or with VS Code

**Pull request merged**